### PR TITLE
fix: deplace FilterDefinition in a file

### PR DESCRIPTION
**Pull request by yannick-aneo**

### CODE REVIEW

File: src/app/types/filter-definition.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+export type FilterDefinitionTaskOption<T extends number | null> = FilterDefinitionTaskOptionString<T>;
+
+export type FilterDefinition<T extends number, U extends number | null = null> = FilterDefinitionRoot<T> | FilterDefinitionTaskOption<U>;
+
+export type FilterFor<T extends number, U extends number | null = null> = FilterDefinition<T, U>['for'];
+
+
+
+export abstract class BasicFilterService {
```
Comment (Faust1-2was-Aneo) [inline src/app/types/filter-definition.ts:82]: DataFilterService might be a better name for this class.

After all, it handles everything data related to the filters, reseting them, restoring them...

File: src/app/sessions/types.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import {SessionRaw as GrpcSessionRaw, SessionRawEnumField, SessionTaskOptionEnumField, SessionTaskOptionGenericField } from '@aneoconsultingfr/armonik.api.angular';
 import { TaskOptions } from '@app/tasks/types';
 import { ColumnKey, FieldKey } from '@app/types/data';
+import { FilterDefinition, FilterFor } from '@app/types/filter-definition';
 import { Filter, FiltersOr } from '@app/types/filters';
 import { ListOptions } from '@app/types/options';
-import { FilterDefinition, FilterFor } from './services/sessions-filters.service';
+
```
Comment (Faust1-2was-Aneo) [inline src/app/sessions/types.ts:7]:
```suggestion
```

Carreful with the empty lines !

File: src/app/types/filter-definition.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+   */
+  for: 'root';
+  field: T;
+  type: 'date';
+};
+
+  
+
+
+export type FilterDefinitionTaskOptionString<T extends number | null> = {
```
Comment (Faust1-2was-Aneo) [inline src/app/types/filter-definition.ts:61]:
```suggestion
};

export type FilterDefinitionTaskOptionString<T extends number | null> = {
```

File: src/app/types/filter-definition.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  for: 'options';
+  field: T;
+  type: 'string';
+};
+
+
+type FilterDefinitionRoot<T extends number> = FilterDefinitionRootString<T> | FilterDefinitionRootNumber<T> | FilterDefinitionRootArray<T> | FilterDefinitionRootStatus<T> | FilterDefinitionRootDate<T>;
+
+
+export type FilterDefinitionTaskOption<T extends number | null> = FilterDefinitionTaskOptionString<T>;
```
Comment (Faust1-2was-Aneo) [inline src/app/types/filter-definition.ts:74]:
```suggestion
type FilterDefinitionRoot<T extends number> = FilterDefinitionRootString<T> | FilterDefinitionRootNumber<T> | FilterDefinitionRootArray<T> | FilterDefinitionRootStatus<T> | FilterDefinitionRootDate<T>;

export type FilterDefinitionTaskOption<T extends number | null> = FilterDefinitionTaskOptionString<T>;
```

**Review by Faust1-2was-Aneo — CHANGES_REQUESTED** [review]

File: src/app/types/filter-definition.ts (hunk 1 above)
Comment (yannick-aneo) [reply src/app/types/filter-definition.ts:78]: I agree with you I'll rename it

**Review by Faust1-2was-Aneo — APPROVED** [review]

**Review by Faust1-2was-Aneo — APPROVED** [review]

**Pull request merged**